packages/cli/src/cmds/navie.ts

```diff
@@ -144,8 +150,15 @@ export function buildNavieProvider(argv: ExplainArgs) {
   ) => {
     loadConfiguration(false);
     const navie = new LocalNavie(contextProvider, projectInfoProvider, helpProvider);
+
     if (argv.threadId) navie.setThreadId(argv.threadId);
+    if (argv.trajectoryFile) {
+      const trajectory = new Trajectory(argv.trajectoryFile);
+      navie.setTrajectoryHandler(trajectory);
 
+      process.on('SIGINT', () => trajectory.close());
+      process.on('exit', () => trajectory.close());
```

Comment:
Two things here:
1.) Binding a `SIGINT` listener will cause the default `SIGINT` handler to unbind itself. The process will no longer via this signal exit unless `process.exit` is called from within this listener.
2.) These listeners will be bound for every message sent. After 10 listeners, node will begin emitting warnings to the console about a potential memory leak. Consider binding these just once.

Reply:
OK, I think the exit handler is all we need. This will only be called once because it's a CLI entry function handler.